modules/sdk/gradle-plugins-jsdoc/src/main/java/com/liferay/gradle/plugins/jsdoc/AppJSDocPlugin.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.gradle.plugins.jsdoc;
+
+import com.liferay.gradle.plugins.node.NodePlugin;
+import com.liferay.gradle.plugins.node.tasks.DownloadNodeModuleTask;
+import com.liferay.gradle.util.GradleUtil;
+
+import groovy.lang.Closure;
+
+import java.io.File;
+
+import java.util.Iterator;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.file.SourceDirectorySet;
+import org.gradle.api.invocation.Gradle;
+import org.gradle.api.plugins.BasePlugin;
+import org.gradle.api.plugins.JavaBasePlugin;
+import org.gradle.api.plugins.PluginContainer;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.bundling.Jar;
+
+/**
+ * @author Andrea Di Giorgi
+ * @author Peter Shin
+ */
+public class AppJSDocPlugin implements Plugin<Project> {
+
+	public static final String APP_JSDOC_TASK_NAME = "appJSDoc";
+
+	public static final String DOWNLOAD_APP_JSDOC_TASK_NAME =
+		"downloadAppJSDoc";
+
+	public static final String JAR_APP_JSDOC_TASK_NAME = "jarAppJSDoc";
+
+	public static final String PLUGIN_NAME = "appJSDocConfiguration";
+
+	@Override
+	public void apply(Project project) {
+		GradleUtil.applyPlugin(project, NodePlugin.class);
+
+		final AppJSDocConfigurationExtension appJSDocConfigurationExtension =
+			GradleUtil.addExtension(
+				project, PLUGIN_NAME, AppJSDocConfigurationExtension.class);
+
+		final JSDocTask appJSDocTask = _addTaskAppJSDoc(project);
+
+		_addTaskJarAppJSDoc(appJSDocTask);
+
+		_configureTaskAppJSDoc(appJSDocTask, _addTaskDownloadAppJSDoc(project));
+
+		Gradle gradle = project.getGradle();
+
+		gradle.afterProject(
+			new Closure<Void>(project) {
+
+				@SuppressWarnings("unused")
+				public void doCall(Project subproject) {
+					Set<Project> subprojects =
+						appJSDocConfigurationExtension.getSubprojects();
+
+					if (subprojects.contains(subproject) &&
+						_hasJSDocSource(subproject)) {
```

Comment:
Here you're checking if the subproject contains a `src/main/resources/META-INF/resources`, and you're also assuming that the subprojects has the `JSDocPlugin` applied. This is true for our projects in `modules` (since `gradle-plugins-defaults` applies the plugin everywhere), but not necessarily for 3rd-party projects. Instead of the `_hasJSDocSource` method, I would simply check here if the subproject has the `JSDocPlugin` applied (look for usages of the `PluginManager` class in other plugins), and maybe filter out non-existing source dirs in `__configureTaskAppJSDoc`.